### PR TITLE
GitHub Actions: Build and Deploy to GitHub Pages.

### PR DESCRIPTION
**Pull request by manjotsidhu**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      # Use GitHub Actions' cache to shorten build times and decrease load on servers
+      - uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Build
+        uses: helaili/jekyll-action@v2
```
Comment (manjotsidhu) [inline .github/workflows/deploy.yml:24]: We can't use this plugin to deploy as in future we need to add a step for Python script to generate APIs.

Comment (satu0king) [reply .github/workflows/deploy.yml:24]: jekyl serve is anyway a separate action and has nothing to do with jekyl build right? So we can use this step independently.

Comment (manjotsidhu) [reply .github/workflows/deploy.yml:24]: True. Let's deploy using only this action. We will keep the python script stuff in a different workflow. Sounds Good ?

**Comment by manjotsidhu** [comment]
Hmmm, I was about to minimalize the script.

**Pull request merged**